test/Conversion/StandardToHandshake/test_remove_redundancies.mlir

```diff
@@ -0,0 +1,61 @@
+// NOTE: Assertions have been autogenerated by utils/update_mlir_test_checks.py
+// RUN: circt-opt -handshake-remove-redundancies %s | FileCheck %s
+module {
+  handshake.func @simple_loop(%arg0: none, ...) -> none {
+    // CHECK-NOT: %0 = "handshake.branch"(%arg0) {control = true} : (none) -> none
+    %0 = "handshake.branch"(%arg0) {control = true} : (none) -> none
+
+    // CHECK: %0:2 = "handshake.control_merge"(%arg0) {control = true} : (none) -> (none, index)
+    %1:2 = "handshake.control_merge"(%0) {control = true} : (none) -> (none, index)
+    %2:3 = "handshake.fork"(%1#0) {control = true} : (none) -> (none, none, none)
+    "handshake.sink"(%1#1) : (index) -> ()
+    %3 = "handshake.constant"(%2#1) {value = 1 : index} : (none) -> index
+    %4 = "handshake.constant"(%2#0) {value = 42 : index} : (none) -> index
+
+    // CHECK-NOT: %5 = "handshake.branch"(%2#2) {control = true} : (none) -> none
+    // CHECK-NOT: %6 = "handshake.branch"(%3) {control = false} : (index) -> index
+    // CHECK-NOT: %7 = "handshake.branch"(%4) {control = false} : (index) -> index
+    %5 = "handshake.branch"(%2#2) {control = true} : (none) -> none
+    %6 = "handshake.branch"(%3) {control = false} : (index) -> index
+    %7 = "handshake.branch"(%4) {control = false} : (index) -> index
+
+    // CHECK: %4 = "handshake.mux"(%7#1, %trueResult, %3) : (index, index, index) -> index
```

Comment:
should probably regex match on the %4 and %7.

Reply:
I'll fix them.